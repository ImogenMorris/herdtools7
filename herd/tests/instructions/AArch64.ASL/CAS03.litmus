AArch64 CAS03
(*
 * This test was failing, because the 32bits memory
 * reads was returning a bitvector of size 64, which
 * compared negatively with the bitvector of size 32
 * read from the register W0.
 *)  
{
int x;
0:X3=x;
}
  P0            ;
 MOV W2,#1      ;
 CAS W0,W2,[X3] ;
forall [x]=1 