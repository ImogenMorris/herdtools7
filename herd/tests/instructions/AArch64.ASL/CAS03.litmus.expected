Test CAS03 Required
States 1
[x]=1;
Ok
Witnesses
Positive: 1 Negative: 0
Condition forall ([x]=1)
Observation CAS03 Always 1 0
Hash=337d3b9628962897752d5dd8bce6694a

